apps/src/StudioApp.js

```diff
@@ -2684,3 +2685,25 @@ StudioApp.prototype.polishGeneratedCodeString = function (code) {
     return code;
   }
 };
+
+/**
+ * Sets a bunch of common page constants used by all of our apps in our redux
+ * store based on our app options config.
+ * @param {AppOptionsConfig}
+ */
+StudioApp.prototype.setCommonPageConstants = function (config) {
```

Comment:
I'm leaning toward the alternative you propose here, leaving `appSpecificConstants` conveniently optional for the apps that don't need to pass anything.  The reason is a little frivolous - I've considered a few times adjusting the `setPageConstants` action to complain if it happens more than once.